src/csharp/Microsoft.Spark/Sql/Types/ComplexTypes.cs

```diff
@@ -241,7 +240,7 @@ public sealed class StructType : DataType
         /// <param name="fields">A collection of StructFieldType objects</param>
         public StructType(IEnumerable<StructField> fields)
         {
-            Fields = fields.ToList();
+            Fields = new List<StructField>(fields);
```

Comment:
@AFFogarty I also have the same question as @suhsteve. I see that the extension method is called, which is doing `return new List<TSource>(source);` Any thought?